File: lib/src/main/java/com/darklycoder/download/info/DownloadConfig.java
package com.darklycoder.download.info;

import android.os.Environment;

import java.io.File;

/**
 * 设置信息
 */
public class DownloadConfig {

    /**
     * 临时的下载目录
     */
    public String tmpPath = getDownloadTmpPath();

    private String getDownloadTmpPath() {
        return Environment.getExternalStorageDirectory().getAbsolutePath() + File.separator + ".downloadTmp";
    }

}

File: lib/src/main/java/com/darklycoder/download/info/DownloadConfig.kt
package com.darklycoder.download.info

import android.os.Environment

import java.io.File

/**
 * 设置信息
 */
class DownloadConfig {

    /**
     * 核心线程池大小
     */
    var corePoolSize = 1

    /**
     * 线程池大小
     */
    var maxPoolSize = 10

    /**
     * 临时的下载目录
     */
    var tmpPath = downloadTmpPath

    init {
        maxPoolSize = Runtime.getRuntime().availableProcessors() + 1
    }

    private val downloadTmpPath: String
        get() = Environment.getExternalStorageDirectory().absolutePath + File.separator + ".downloadTmp"

}
